tensorflow-framework/src/main/java/org/tensorflow/framework/activations/ReLU.java

```diff
@@ -0,0 +1,143 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+=======================================================================*/
+package org.tensorflow.framework.activations;
+
+import org.tensorflow.DataType;
+import org.tensorflow.Operand;
+import org.tensorflow.op.Ops;
+import org.tensorflow.op.math.Greater;
+import org.tensorflow.op.nn.LeakyRelu;
+import org.tensorflow.types.family.TNumber;
+
+/**
+ * Rectified Linear Unit(ReLU) activation.
+ *
+ * <p>With default values, this returns the standard ReLU activation: <code>max(x, 0)</code>, the
+ * element-wise maximum of 0 and the input tensor.
+ *
+ * <p>Modifying default parameters allows you to use non-zero thresholds, change the max value of
+ * the activation, and to use a non-zero multiple of the input for values below the threshold.
+ *
+ * <p>For example:
+ *
+ * <pre>
+ *     Operand&lt;TFloat32&gt; input = tf.constant(
+ *              new float[] {-10f, -5f, 0.0f, 5f, 10f});
+ *
+ *     // With default parameters
+ *     ReLU&lt;TFloat32&gt; relu = new ReLU&lt;&gt;(tf);
+ *     Operand&lt;TFloat32&gt; result = relu.call(input);
+ *     // result is [0.f,  0.f,  0.f,  5.f, 10.f]
+ *
+ *     // With alpha = 0.5
+ *     relu = new ReLU&lt;&gt;(tf, 0.5f, ReLU.MAX_VALUE_DEFAULT, ReLU.THRESHOLD_DEFAULT);
+ *     result = relu.call(input);
+ *     // result is [-5.f , -2.5f,  0.f ,  5.f , 10.f]
+ *
+ *     // With maxValue = 5
+ *     relu = new ReLU&lt;&gt;(tf, ReLU.ALPHA_DEFAULT, 5f, ReLU.THRESHOLD_DEFAULT);
+ *     result = relu.call(input);
+ *     // result is [0.f, 0.f, 0.f, 5.f, 5.f]
+ *
+ *     // With threshold = 5
+ *     relu = new ReLU&lt;&gt;(tf, ReLU.ALPHA_DEFAULT, ReLU.MAX_VALUE_DEFAULT, 5f);
+ *     result = relu.call(input);
+ *     // result is [-0.f, -0.f,  0.f,  0.f, 10.f]
```

Comment:
How does this emit negative zero? Could we catch that and emit a positive zero?

Reply:
I am not sure who is the culprit here. It seems to be the result of the multiply operation of a negative number * zero always produces `-0`, maybe this is a Pyhtonism.
Here is the same operation in just `Numpy`:
```
a = np.array([-10., -5., 0.0, 5., 10.], dtype=np.float32)
c = np.greater(a, 5.)
print(c);
[False False False False  True]

c = np.cast['float32'](c)
print(c)
[0. 0. 0. 0. 1.]

d = a * c
print(d)
[-0. -0.  0.  0. 10.]

z = -1. * 0.
print(z)
-0.0

```
TensorFlow Python and TensorFlow Java produce the same `-0 `result.

If you don't want this behavior, we could add zero to the final result, because `-0 + 0 = 0`.

Reply:
@Craigacp , do you have any other comment about this?

Reply:
If TF python is doing it then I guess it's fine.